KubeArmor/core/hook_handler.go

```diff
@@ -0,0 +1,200 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of KubeArmor
+
+package core
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"net"
+	"os"
+	"path/filepath"
+	"sync/atomic"
+
+	kl "github.com/kubearmor/KubeArmor/KubeArmor/common"
+	cfg "github.com/kubearmor/KubeArmor/KubeArmor/config"
+	"github.com/kubearmor/KubeArmor/KubeArmor/types"
+)
+
+const kubearmorDir = "/var/run/kubearmor"
+
+// ListenToHook starts listening on a UNIX socket and waits for container hooks
+// to pass new containers
+func (dm *KubeArmorDaemon) ListenToHook() {
+	if err := os.MkdirAll(kubearmorDir, 0750); err != nil {
+		log.Fatal(err)
+	}
+
+	listenPath := filepath.Join(kubearmorDir, "ka.sock")
+	_ = os.Remove(listenPath) // in case kubearmor crashed and the socket wasn't removed
+	socket, err := net.Listen("unix", listenPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer socket.Close()
+	defer os.Remove(listenPath)
+	ready := &atomic.Bool{}
+
+	for {
+		conn, err := socket.Accept()
+		if err != nil {
+			log.Fatal(err)
```

Comment:
The reason for some of the fatal error handling is because if we just logged that there is an error and continued, some containers will might go unnoticed for the duration of KubeArmor lifetime. Do you think it's better if we leave KubeArmor running and not monitoring some containers or restart KubeArmor and get all containers from the start (using some form of exiting)?